src/commands/push.ts

```diff
@@ -95,5 +95,7 @@ const manifestHasChanges = async (): Promise<boolean> => {
   const remoteFiles = await fetchProject(scriptId, undefined, true);
   const remoteManifest = remoteFiles.find((file) => file.name === PROJECT_MANIFEST_BASENAME);
   if (!remoteManifest) throw Error('remote manifest no found');
-  return localManifest !== remoteManifest.source;
+  return localManifest !== remoteManifest.source
+    // if string comparison fail, double check after 'normalization'
+    && JSON.stringify(JSON.parse(localManifest)) !== JSON.stringify(JSON.parse(remoteManifest.source));
```

Comment:
Don't use JSON parse.

See https://github.com/google/clasp/pull/431#discussion_r237969173

Reply:
I had to add it as a failsafe. For some reason, I had a project where the string comparison was always failing. Adding the (silly) JSON stringify & parse allowed to workaround the issue.

I will remove it and re-test my projects.